test-e2e/tests/payments/subscription_test.ts

```diff
@@ -144,7 +143,7 @@ function runTestSuite(props: ProviderProps, providerName: string) {
   });
 
   Scenario(`I can open the PayPal site - ${providerName}`, async ({ I }) => {
-    paidLoginContext = await I.registerOrLogin(paidLoginContext);
+    await I.registerOrLogin();
```

Comment:
The extra runtime and junk data is imo justified by the modularity and readability gained by having all tests actually be e2e, and it cuts down on future development time. If one of these scenarios changes and you have to change the test, you don't need to wait for 4 tests to run with all the necessary setup just to get to the one test that you're working on. Just my 2c

Reply:
Any of these tests can be run independently or in any order:
- `I can see my payments data `
- `I can see offered subscriptions`
- `I can choose an offer`
- `I can see payment types`
- `I can open the PayPal site`

These tests are the 4 that depend on each other
- `I can finish my subscription with credit card`
- `I can cancel my subscription`
- `I can renew my subscription`
- `I can view my invoices`

We could probably combine those 4 or rewrite them in a way to better support 1-off runs if necessary, but you'd either lose resolution on test results, or still have the same problem that a bug in subscription creation also causes the cancel test to fail, etc. But in this case, since you've made it run the same flow 4 times, you've also added 2 minutes to the total runtime.